server/game/gameSystems/PlayCardSystem.ts

```diff
@@ -33,15 +40,37 @@ export class PlayCardSystem<TContext extends AbilityContext = AbilityContext> ex
         ignoredRequirements: [],
         optional: false,
         entersReady: false,
-        playType: PlayType.PlayFromHand
+        playType: PlayType.PlayFromHand,
+        nested: false
     };
 
     public eventHandler(event, additionalProperties): void {
         const player = event.player;
         const newContext = (event.playCardAbility as PlayCardAction).createContext(player);
+
+        /* if (this.properties.costAdjusterProperties) {
+            this.queueApplyCostAdjusterGameSteps(this.properties.costAdjusterProperties, newContext);
+        }*/
         event.context.game.queueStep(new AbilityResolver(event.context.game, newContext, event.optional));
     }
 
+    private queueApplyCostAdjusterGameSteps(costAdjusterProperties: ICostAdjusterProperties, context: AbilityContext) {
+        const applyCostAdjusterSystem = new PlayerLastingEffectSystem({
+            duration: Duration.UntilEndOfAction,
+            effect: modifyCost(Object.assign(costAdjusterProperties, { limit: AbilityLimit.perGame(1), playingTypes: context.playType }))
```

Comment:
nit: a little cleaner to use object assignment syntax instead of Object.assign imo - e.g. `{ ...costAdjusterProperties, limit: AbilityLimit.perGame(1), playingTypes: context.playType }`